### PR TITLE
[WIP] [Payum] Adding offline payment method 

### PR DESCRIPTION
**Pull request by antonioperic**
Hi, 
I suppose that offline payment method should be used as Wire/Check payment so I think it will be good to have it. 
I configure and added it but there is one slight issues. I think this payment should me marked as payed = false, but default FillOrderDetails always override my configuration. How to change this @makasim ? Should we add some if statement to core of Payum/Offline or?

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml [hunk 1]
```diff
@@ -152,6 +152,7 @@ sylius_payment:
             controller: Sylius\Bundle\CoreBundle\Controller\PaymentController
     gateways:
         dummy: Test
+        offile: Offline
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius.yml]: Typo.

**Comment by stloyd** [comment]
You should not commit `composer.phar`...

**Comment by antonioperic** [comment]
@stloyd that was mistake with composer.phar I will clean it at the end and squash. Just want to see how to setup this paid = false cc @makasim

File: src/Sylius/Bundle/PayumBundle/Payum/Offline/Action/CapturePaymentAction.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+     * @throws LogicException
+     */
+    protected function composeDetails(PaymentInterface $payment, TokenInterface $token)
+    {
+        if ($payment->getDetails()) {
+            return;
+        }
+
+        $details = array();
+        $details[Constants::FIELD_PAID] = false;
```
Comment (makasim) [inline src/Sylius/Bundle/PayumBundle/Payum/Offline/Action/CapturePaymentAction.php]: That's correct if you want to create a pending payment. @antonioperic  Is it your case ?

Comment (antonioperic) [inline src/Sylius/Bundle/PayumBundle/Payum/Offline/Action/CapturePaymentAction.php]: @makasim my idea was to use offline as wire/Cheque payment. That is perfect solution. But in this case if you complete your order payment is PENDING not completed.

Comment (antonioperic) [inline src/Sylius/Bundle/PayumBundle/Payum/Offline/Action/CapturePaymentAction.php]: And this doesn't work. It is always marked as PAID as default offline method overrides this. There is no
    if($payment->getDetails())....

https://github.com/Payum/Offline/blob/master/Action/FillOrderDetailsAction.php#L18

I can PR this if you want

Comment (makasim) [inline src/Sylius/Bundle/PayumBundle/Payum/Offline/Action/CapturePaymentAction.php]: Yes, it has to be fixed in Offline, we should not force it.

**Comment by antonioperic** [comment]
@makasim can now this be closed?

**Comment by makasim** [comment]
looks good to me

**Comment by antonioperic** [comment]
cc @stloyd

**Comment by stloyd** [comment]
Just that typo, but in overall looks good.

**Comment by pjedrzejewski** [comment]
Requires a rebase.

**Comment by aramalipoor** [comment]
@antonioperic @makasim What's the status of this PR?

**Comment by antonioperic** [comment]
@aramalipoor this is completed and merged

**Comment by aramalipoor** [comment]
@antonioperic @stloyd Great job! U rock!

**Pull request closed without merging**